app/assets/javascripts/omni-nav-v2.js

```diff
@@ -1,38 +1,109 @@
-      $('html.omni-nav-v2').toggleClass('utility-nav-open');
-    }else {
-      omni_nav_v2.$primary_nav.toggleClass('search-open');
+      $utility_nav.toggleClass('utility-nav-open');
+      $('html.omni-nav-v2').toggleClass('utility-nav-open');  
+      $primary_nav.removeClass('search-open');    
+    } else {
+      $utility_nav.removeClass('utility-nav-open');
+      $('html.omni-nav-v2').removeClass('utility-nav-open'); 
+      $primary_nav.toggleClass('search-open');
+    }
+
+    var primary_nav_classes = document.getElementById('primary-nav').classList;
+
+    // jQuery < 3.0 doesn't support toggleClass for SVGs
+    // toggle open iff one of the search bars is open
+    if(primary_nav_classes.contains("search-open") || $utility_nav.hasClass('utility-nav-open')) {
+      $('.icon-open-search').attr("class", "icon-open-search hide");
+      $('.icon-close-search').attr("class", "icon-close-search");
+    } else {
+      $('.icon-open-search').attr("class", "icon-open-search");
+      $('.icon-close-search').attr("class", "icon-close-search hide");
     }
-  },
+  };
 
-  onUtilityNavDropdownClick : function(e) {
+  var onUtilityNavDropdownClick = function(e) {
     e.stopPropagation();
     $(this).siblings('.utility-has-dropdown').removeClass('dropdown-open');
     $(this).toggleClass('dropdown-open');
-    $(document).on("click", omni_nav_v2.onDocumentClick);
-  },
+    $(document).on("click", onDocumentClick);
+  };
 
-  onDocumentClick : function(e){
+  var onDocumentClick = function() {
     $('li.utility-has-dropdown').removeClass('dropdown-open');
-    $(document).off("click", omni_nav_v2.onDocumentClick);
-  }
-}
+    $(document).off("click", onDocumentClick);
+  };
+
+  var onSearchInput = function() {
+    $('.search-icon').addClass('hide');
+    $(document).on("click", function() {
+      // resets search input box when user clicks outside
+      $utility_nav.find('input:text').val("");
+      $primary_nav.find('input:text').val("");
+      $('.search-icon').removeClass('hide');
+    });
+  };
+
+  return {
+    init: initialize
+  };
+})();
+
+var OffCanvasNav = (function() {
+
+  var initialize = function() {
+    syncLinkWidths();
+
+    $('#js-off-canvas-trigger, #js-close-off-canvas-nav, #js-off-canvas-overlay').on('click', function(event) {
+      event.preventDefault();
+      $('#js-off-canvas-nav-container').toggleClass('open');
+      $('#js-off-canvas-overlay').toggleClass('active');
+      $('body').toggleClass('no-scroll');
+    });
+
+    $('#js-off-canvas-nav-container .toggle').on('click', function() {
+      $(this).parent().toggleClass('open'); // Targets li
+      $(this).parent().find('ul').slideToggle();
+    });
+
+    $(window).on('resize', function() {
+      syncLinkWidths();
+    });
+  };
+
+  var syncLinkWidths = function() {
+    var width = $('#js-off-canvas-nav > ul').width();
+    $('#js-off-canvas-nav > ul > li > a').css('width', width);
```

Comment:
I would create a variable for `$('#js-off-canvas-nav > ul > li > a')` in the `initialization` method so you don't have to parse the DOM each time`syncLinkWidths` is called.